docs/UsersGuide.asciidoc

```diff
@@ -1109,6 +1113,72 @@ timeouts by performing the task in background.
 This is recommended on big instances but means that the results (and
 possible errors) need to be polled via `openqa-cli api isos/$scheduled_product_id`.
 
+[id="scenarios_yaml"]
+===== Defining test scenarios in YAML =====
+Instead of relying on the tables for machines, mediums/products, test suites and
+job templates of the openQA instance, one can provide these definitions/settings
+also via a YAML document. This YAML document could be specific to a certain test
+distribution and stored in the same repository as those tests (making the versioning
+easier).
+
+WARNING: This feature is still experimental and may change in an incompatible way
+in future versions.
+
+This YAML document can be specified via the scheduling parameter
+`SCENARIO_DEFINITIONS_YAML`:
+
+[source,sh]
+--------------------------------------------------------------------------------
+openqa-cli api … -X POST isos --param-file SCENARIO_DEFINITIONS_YAML=/local/file.yaml …
+--------------------------------------------------------------------------------
+
+This command will upload the contents of the local file `/local/file.yaml` to a
+possibly remote openQA instance. The YAML document will only be used within the
+scope of this particular API request. No settings are stored/altered on the openQA
+instance.
+
+If the YAML document already exists on the openQA host, you can also use
+`SCENARIO_DEFINITIONS_YAML_FILE` which expects the file path of the YAML document
+on the openQA host.
+
+The YAML document itself should define at least one product, machine and job
+template like this:
```

Comment:
As discussed 2023-03-09 we shouldn't see product+machine as mandatory. With *just* job templates defining all variables that effectively makes them "scenarios". Maybe we can have as first example the most simple scenario definition and after that provide the optional products+machines extensions

Reply:
This is not yet possible. If it should already be optional in the first merged version I can work on that.

Reply:
I'd prefer that